samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemModifyListener.java

```diff
@@ -748,9 +748,11 @@ private void populateItemTextForCalculatedQuestion(ItemAuthorBean itemauthorbean
               for (AnswerIfc answer : answers) {
                   if (answer.getIsCorrect()) {
                       String text = answer.getText();
-                      String min = text.substring(0, text.indexOf("|"));
-                      String max = text.substring(text.indexOf("|") + 1, text.indexOf(","));
-                      String decimalPlaces = text.substring(text.indexOf(",") + 1);              
+                      String[] partsText = text.split("\\|");
+                      String min = partsText[0];
```

Comment:
Does the answer text always have a pipe separator? Do we need a length check on this array?